src/client/submit.rs

```diff
@@ -0,0 +1,53 @@
+use crate::errors;
+
+use async_trait::async_trait;
+use errors::{Error, Result};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Submission {
+    #[structopt(short, long)]
+    pub problem_id: Option<String>,
+
+    #[structopt(short, long, default_value = "3")]
+    pub lang_id: String,
+
+    #[structopt(parse(from_os_str))]
+    pub file: PathBuf,
```

Comment:
Maybe, replace `pub` with the `get_source` function? Could be improved for other streams, like stdin/out